db/cart.js

```diff
@@ -0,0 +1,93 @@
+const client = require("./client");
+
+
+async function addToCartItems({
+    userId,
+    puppyId,
+}) {
+    try{
+        const { rows: [cart_item] } = await client.query(`
+            INSERT INTO cart_items("userId", "puppyId")
+            VALUES ($1, $2)
+            ON CONFLICT ("userId", "puppyId") DO NOTHING
+            RETURNING *;
+        `, [userId, puppyId])
+
+    return cart_item;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function getCartById(id) {
+try{
+        const { rows: [ cart_item ] } = await client.query(`
+            SELECT *
+            FROM cart_items
+            WHERE id=$1;
+        `, [id])
+
+        return cart_item;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function getCartByUser({ id }) {
+    try{
+        const { rows } = await client.query(`
+            SELECT *
+            FROM cart_items
+            WHERE "userId"=$1;
+        `, [id])
+
+        return rows;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function updateCartItems({ id, ...fields }) {
+try{
+        const setString = Object.keys(fields).map(
+        (key, index) => `"${ key }"=$${ index + 1 }`
+        ).join(', ');
+
+        if (setString.length === 0) {
+            return;
+        }
+
+        const { rows: [ cart_item ] } = await client.query(`
+            UPDATE cart_items
+            SET ${ setString }
+            WHERE id=${id}
+            RETURNING *;
+        `, Object.values(fields));
+
+        return cart_item;
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+async function deleteCartItems(id) {
```

Comment:
I think we might want a deleteCart(userId) function as well so we can just clear out a users cart when they do a checkout.

Reply:
done